merino-suggest/src/lib.rs

```diff
@@ -336,110 +337,46 @@ pub enum SuggestError {
 
 /// Languages supported by the client.
 #[derive(Debug, PartialEq)]
-pub struct SupportedLanguages(pub Vec<Language>);
+pub struct SupportedLanguages(pub AcceptLanguage);
 
 impl SupportedLanguages {
-    /// Create a new SupportedLanguages instance with a wildcard that has no quality value.
-    pub fn wildcard() -> Self {
-        let language = Language {
-            language_identifier: LanguageIdentifier::Wildcard,
-            quality_value: None,
-        };
-
-        Self(vec![language])
-    }
-
     /// Specify whether `self` includes the language specified by the given language and region.
     pub fn includes(&self, language_query: &str, region_query: Option<&str>) -> bool {
-        let region_matches = |supported_region| {
-            match (supported_region, region_query) {
-                // If the region query is None, the caller intends to match every region
-                (_, None) => true,
-                // If the region query is Some(_) but the supported region is None, the regions
-                // don't match
-                (None, Some(_)) => false,
-                (Some(supported_region), Some(region_query)) => supported_region == region_query,
+        let language_tag_to_match = {
```

Comment:
Would it be preferable for this function just to accept a single `LanguageTag` as its argument?

Reply:
I think so, yes. That better exposes the error case of if `language_query` and `region_query` are invalid. That better follows the "parse don't verify" pattern.